app/discovery/discovery.go

```diff
@@ -146,6 +148,37 @@ func (s *Service) Match(srv, src string) (string, MatchType, bool) {
 	return src, MTProxy, false
 }
 
+// ScheduleHealthCheck starts background loop with health-check
+func (s *Service) ScheduleHealthCheck(ctx context.Context, interval time.Duration) {
+	log.Printf("health-check scheduled every %s", interval)
+
+	go func() {
+	hloop:
+		for {
+			timer := time.NewTimer(interval)
+			select {
+			case <-timer.C:
+				s.lock.RLock()
+				cres := CheckHealth(s.Mappers())
+				s.lock.RUnlock()
+
+				// alive services would be picked up first
+				sort.SliceStable(cres.mappers, func(i, j int) bool {
+					return cres.mappers[j].dead
+				})
+				s.lock.Lock()
+				s.mappers = make(map[string][]URLMapper)
+				for _, m := range cres.mappers {
+					s.mappers[m.Server] = append(s.mappers[m.Server], m)
```

Comment:
unless I'm blind I can't see where dead flag used latter on

Reply:
I used flag to sort the slice with destinations. See my answer above.